lib/rubydoop/dsl.rb

```diff
@@ -425,12 +428,16 @@ def wait_for_completion(verbose)
       end
 
       class Parallel < Jobs
+        def submit
+        end
+
         def wait_for_completion(verbose)
+          @jobs.each do |job|
+            job.submit
+          end
           @jobs.map do |job|
-            Thread.new do
-              job.wait_for_completion(verbose)
-            end
-          end.map!(&:value).all?
+            job.wait_for_completion(verbose)
+          end.all?
```

Comment:
You still need the threads here as subsequent jobs in sequences are otherwise not started appropriately. Concretely, for the following case:

```ruby
parallel do
  sequence { a=job { ... }; b = job {...} }
  sequence { c=job { ... }; d = job {...} }
end
```

The job `d` will not be submitted until `b` has completed without threads, with threads `b` and `d` could execute in parallel.